<launch>
	<group ns="fmSensors">
		<node pkg="plot_propulsion_feedback_leftright" name="plot_feedback" type="plot_propulsion_feedback_leftright_node.py" output="screen" >
 			<param name="propulsion_feedback_left_sub" value="/fmInformation/propulsion_module_feedback_left" />
 			<param name="propulsion_feedback_right_sub" value="/fmInformation/propulsion_module_feedback_right" />
 			<param name="update_frequency" value="10" />
			<param name="plot_title" value="Wheel feedback" />
			<param name="maximum_velocity" value="1.2" />
			<param name="maximum_thrust" value="540" />
		</node>

		<node pkg="sdu_armadillo3_test" name="pid_left" type="plot_pid_node.py" output="screen" >
 			<param name="pid_sub" value="/fmInformation/pid_left" />
 			<param name="update_rate" value="10" />
			<param name="plot_title" value="PID state" />
			<param name="samples" value="500" />
			<param name="max_error" value="1.0" />
			<param name="max_output" value="1.0" />
			<param name="max_pid" value="1.0" />
			<param name="plot_feed_forward" value="false" />
		</node>
		
		<node pkg="sdu_armadillo3_test" name="pid_right" type="plot_pid_node.py" output="screen" >
 			<param name="pid_sub" value="/fmInformation/pid_right" />
 			<param name="update_rate" value="10" />
			<param name="plot_title" value="PID state" />
			<param name="samples" value="500" />
			<param name="max_error" value="0.1" />
			<param name="max_output" value="0.1" />
			<param name="max_pid" value="0.1" />
			<param name="plot_feed_forward" value="false" />
		</node>
	</group>
	
	<group ns="fmDecisionMakers">
		<node pkg="sdu_armadillo3_test" name="mission" type="keyboard_mission_node.py" output="screen">
			<param name="keyboard_sub" value="/fmHMI/keyboard" />
			<param name="automode_pub" value="/fmDecision/automode" />
    		<param name="deadman_pub" value="/fmCommand/deadman" />
			<param name="cmd_vel_pub" value="/fmCommand/cmd_vel" />
            <param name="max_linear_velocity" value="1.0"/>
            <param name="max_angular_velocity" value="0.6"/>
            <param name="linear_velocity_step" value="0.2"/>
            <param name="angular_velocity_step" value="0.1"/>
		</node>
	</group>



	<group ns="fmLib">
		 <node pkg="keyboard_interface" type="keyboard_node" name="keyboard">
			<param name="keyboard_pub" value="/fmHMI/keyboard" />
		</node>

		<node pkg="differential_ifk" type="differential_ifk_node" name="differential_ifk" output="screen">
			<param name="hl_subscriber_topic" value="/fmCommand/cmd_vel" />
			<param name="ll_publisher_topic_left" value="/fmSignals/cmd_vel_left" />
			<param name="ll_publisher_topic_right" value="/fmSignals/cmd_vel_right" />
			<param name="distance_center_to_wheel" value="1.10" />
			<param name="deadmanbutton_topic" value="/fmCommand/deadman" />
		</node>

		<node pkg="sdu_armadillo3" type="sdu_armadillo3_node" name="roboteq_left" output="screen" respawn="true">
			<param name="pid_topic" value="/fmInformation/pid_left" />
			<param name="cmd_vel_ch1_topic" value="/fmSignals/cmd_vel_left"/>
			<param name="serial_rx_topic" value="/fmData/left_belt_rx" />
			<param name="serial_tx_topic" value="/fmData/left_belt_tx" />
			<param name="deadman_topic" value="/fmCommand/deadman"/>
			<param name="encoder_ch1_topic" value="/fmInformation/encoder_left"/>
			<param name="propulsion_module_status_topic" value="/fmInformation/propulsion_module_status"/>
			<param name="propulsion_module_feedback_topic" value="/fmInformation/propulsion_module_feedback_left"/>
			<param name="power_ch1_topic" value="/fmData/power_left"/>
			<param name="status_topic" value="/fmData/status"/>
			<param name="ticks_per_meter" value="740" />
			<param name="mps_to_thrust" value="500"/>
			<param name="max_controller_command" value="800" />
			<param name="p_gain" value="1.0"/>
			<param name="i_gain" value="3.0"/>
			<param name="d_gain" value="0.1"/>
			<param name="feed_forward" value="1.5"/>
			<param name="i_max" value="0.1"/>
			<param name="closed_loop_operation" value="true"/>
			<param name="position_control" value="false"/>
			<param name="robot_max_acceleration" value="0.5"/>
			<param name="robot_max_jerk" value="0.5"/>
			<param name="brake_zeroband" value="0.2"/>  
			<param name="velocity_tolerance" value="0.05"/>  
		</node>
		<node pkg="sdu_armadillo3" type="sdu_armadillo3_node" name="roboteq_right" output="screen" respawn="true">
			<param name="pid_topic" value="/fmInformation/pid_right" />
			<param name="cmd_vel_ch1_topic" value="/fmSignals/cmd_vel_right"/>
			<param name="serial_rx_topic" value="/fmData/right_belt_rx" />
			<param name="serial_tx_topic" value="/fmData/right_belt_tx" />
			<param name="deadman_topic" value="/fmCommand/deadman"/>
			<param name="encoder_ch1_topic" value="/fmInformation/encoder_right"/>
			<param name="propulsion_module_status_topic" value="/fmInformation/propulsion_module_status"/>
			<param name="propulsion_module_feedback_topic" value="/fmInformation/propulsion_module_feedback_right"/>
			<param name="power_ch1_topic" value="/fmData/power_right"/>
			<param name="status_topic" value="/fmData/status"/>
			<param name="ticks_per_meter" value="740" />
			<param name="mps_to_thrust" value="500"/>
			<param name="max_controller_command" value="800" />
			<param name="p_gain" value="1.0"/>
			<param name="i_gain" value="3.0"/>
			<param name="d_gain" value="0.1"/>
			<param name="feed_forward" value="1.5"/>
			<param name="i_max" value="0.1"/>
			<param name="closed_loop_operation" value="true"/>  
			<param name="position_control" value="false"/>
			<param name="robot_max_acceleration" value="0.5"/>
			<param name="robot_max_jerk" value="0.5"/>
			<param name="brake_zeroband" value="0.2"/>  
			<param name="velocity_tolerance" value="0.05"/>     
		</node>

		<node pkg="serial_string" type="serial_string_node" name="serial_comm_left" output="screen">
			<param name="publisher_topic" value="/fmData/left_belt_rx" />
			<param name="subscriber_topic" value="/fmData/left_belt_tx" />
			<param name="device" value="/dev/belt_left" />
			<param name="baudrate" value="115200" />
			<param name="termination_character" value="13" />
		</node>
	
		<node pkg="serial_string" type="serial_string_node" name="serial_comm_right" output="screen">
			<param name="publisher_topic" value="/fmData/right_belt_rx" />
			<param name="subscriber_topic" value="/fmData/right_belt_tx" />
			<param name="device" value="/dev/belt_right" />
			<param name="baudrate" value="115200" />
			<param name="termination_character" value="13" />
		</node>
	</group>	
</launch>
